main.go

```diff
@@ -114,7 +114,7 @@ func main() {
 	if err != nil {
 		log.Fatalf("failed to start tcp listener: %v\n", err)
 	}
-	opt := grpc.UnaryInterceptor(pb.Interceptor(logger))
+	opt := grpc.UnaryInterceptor(pb.Interceptor(logger, auth.TokenToUserMap))
```

Comment:
This argument to interceptor is not necessary since `auth.TokenToUserMap` is accessible also from inside the interceptor implementation. That said, it might be that we do want to pass something into the interceptor from main if we had created some users session map in main. Right now, it is not clear what is the better solution.

Reply:
On another note. I've been meaning to rewrite the current interceptor, splitting the logic into separate interceptors, and use chaining of the different interceptors. See [ChainUnaryInterceptor](https://pkg.go.dev/google.golang.org/grpc#ChainUnaryInterceptor).